public/app/features/explore/TableContainer.tsx

```diff
@@ -18,27 +20,49 @@ interface TableContainerProps {
   width: number;
   timeZone: TimeZone;
   onCellFilterAdded?: (filter: FilterItem) => void;
+  showRawPrometheus?: boolean;
   splitOpenFn: SplitOpen;
 }
 
+interface TableContainerState {
+  resultsStyle: TableResultsStyle;
+}
+
 function mapStateToProps(state: StoreState, { exploreId }: TableContainerProps) {
   const explore = state.explore;
-  // @ts-ignore
-  const item: ExploreItemState = explore[exploreId];
-  const { loading: loadingInState, tableResult, range } = item;
-  const loading = tableResult && tableResult.length > 0 ? false : loadingInState;
-  return { loading, tableResult, range };
+  const item: ExploreItemState = explore[exploreId]!;
+  const { loading: loadingInState, tableResult, rawPrometheusResult, range } = item;
+  const rawPrometheusFrame: DataFrame[] = rawPrometheusResult ? [rawPrometheusResult] : [];
+  const result = (tableResult?.length ?? false) > 0 && rawPrometheusResult ? tableResult : rawPrometheusFrame;
+  const loading = result && result.length > 0 ? false : loadingInState;
+
+  return { loading, tableResult: result, range };
 }
 
 const connector = connect(mapStateToProps, {});
 
 type Props = TableContainerProps & ConnectedProps<typeof connector>;
 
-export class TableContainer extends PureComponent<Props> {
+export class TableContainer extends PureComponent<Props, TableContainerState> {
```

Comment:
In a previous discussion it was suggested to create a new component the would import the table rather than extending the table itself. why did that change?

we want to avoid the table to become a "dumpster component" for everything that can be eventually rendered as tabular data (~anything), hence `TableContainer` should really only render a container and its table. i'd suggest to move everything else inside another `*Container` component that holds this logic and eventually reuses some bits of this component (if we need to refactor TableContainer to make that happen happy to see that happen as well) pretty much like LogsContainer reuses some bits from the graph panel to show the volume histogram

Reply:
Sounds good, I split this off into a new RawPrometheusContainer element, let me know what you think!